### ISSUE TITLE
Add support for nullable properties

### ISSUE THREAD

**Issue by namelivia** [JsonMapper/JsonMapper#57]
While serializing objects I'm getting some properties set to null sometimes, and this is crashing my application with the following message:
```
Argument 1 passed to JsonMapper\JsonMapper::mapObject() must be an instance of stdClass, null given
```
As `json-mapper/json-mapper/src/JsonMapper.php::mapObject` gets the null value instead of the json object corresponding to the standard response.

I've tried setting the variable[ as nullable](https://docs.phpdoc.org/latest/references/phpdoc/tags/var.html) in the docblock but it seems the whole comment line is taken as a string instead of separated in components.
When I update this:
```php
/**
 * @var TravelperkBankAccount
 */
public $travelperkBankAccount;
```

To this:
```php
/**
 * @var TravelperkBankAccount|null
 */
public $travelperkBankAccount;
```

I get this:
```
Class 'Namelivia\TravelPerk\Expenses\Types\TravelperkBankAccount|null' not found
```
on `json-mapper/json-mapper/src/Handler/PropertyMapper.php::mapPropertyValue`.

To my understanding what comes on the type comment is not being parsed and broken down into components, because I get a similar error too if I write:
```php
/**
 * @var TravelperkBankAccount This is a bank account number.
 */
public $travelperkBankAccount;
```

More or less I think how could this new feature would be implemented, so if you confirm my assumptions are right I can try implementing it myself and opening a PR.

**Comment by DannyvdSluijs** [comment]
@namelivia Thanks for reporting the issue. The library should support nullable properties. I’ll see if I can reproduce somewhere coming week. 

Could you tell me which version of JsonMapper you are using? And if you have share a piece of code which shows the issue.

**Comment by namelivia** [comment]
Yes! I was just digging a bit more and saw [it already had this feature](https://github.com/JsonMapper/JsonMapper/blob/master/src/Middleware/DocBlockAnnotations.php#L89).

I'm currently using [1.4.0](https://github.com/namelivia/travelperk-http-php/blob/master/composer.lock#L231) I'll make a branch wit a test for this and link it here.

**Comment by namelivia** [comment]
My bad! I found the issue while investigating a bit, I wasn't explicitly asking for a `DocBlockAnnotations` middleware, going from:
```php
<?php

declare(strict_types=1);

namespace Namelivia\TravelPerk;

use JsonMapper\Enums\TextNotation;
use JsonMapper\JsonMapperFactory;
use JsonMapper\Middleware\CaseConversion;
use JsonMapper\JsonMapper;

class MapperFactory
{
    public function build(): JsonMapper
    {
        $mapper = (new JsonMapperFactory())->default();
        $mapper->push(new CaseConversion(TextNotation::UNDERSCORE(), TextNotation::CAMEL_CASE()));
        return $mapper;
    }
}
```
to:
```php
<?php

declare(strict_types=1);

namespace Namelivia\TravelPerk;

use JsonMapper\Enums\TextNotation;
use JsonMapper\JsonMapperFactory;
use JsonMapper\Middleware\CaseConversion;
use JsonMapper\Middleware\DocBlockAnnotations;
use JsonMapper\Cache\ArrayCache;
use JsonMapper\JsonMapper;

class MapperFactory
{
    public function build(): JsonMapper
    {
        $mapper = (new JsonMapperFactory())->default();
        $mapper->push(new DocBlockAnnotations(new ArrayCache()));
        $mapper->push(new CaseConversion(TextNotation::UNDERSCORE(), TextNotation::CAMEL_CASE()));
        return $mapper;
    }
}
```
Fixed it, Thanks! :star: , closing this now.

**Comment by namelivia** [comment]
I found the issue, I think the problem lies on the calculation of the type in the DocBlockAnnotations middleware, the nullable calculation is right but the type is what is failing. I've forked the repo and wrote a [regression test that is failing](https://github.com/JsonMapper/JsonMapper/compare/develop...namelivia:fix_docblock_type_parsing). Working on the fix now.

**Comment by DannyvdSluijs** [comment]
@namelivia I've released version 1.4.1 which should solve your issues, see:
- https://github.com/JsonMapper/JsonMapper/releases/tag/1.4.1
- https://packagist.org/packages/json-mapper/json-mapper

If you like JsonMapper please give it a star and follow us on [Twitter](https://twitter.com/JsonMapper)
